quantity.hpp

```diff
@@ -63,6 +71,218 @@ using    day_scaling = std::ratio_multiply<  hour_scaling, scale<24>>;
 using   week_scaling = std::ratio_multiply<   day_scaling, scale< 7>>;
 /** @} */
 
+
+/* tags **********************************************************************/
+
+/**
+ * @brief Quantity tag
+ *
+ * Tagged quantities are tagged with a tag, which consists of an ID and an
+ * exponent. The ID is used to tell tags from each other. Any type, even an
+ * incomplete type, will do for the ID. The exponent is used to track
+ * multiplication and division of tagged quantities so that tags can be
+ * canceled automatically when divided.
+ *
+ * @tparam  ID  tag ID
+ * @tparam Num  tag ratio numerator
+ * @tparam Den  tag ratio denominator
+ *
+ * @note Do not this type directly. Use the create_tag_t meta function instead.
+ */
+template<typename ID, std::intmax_t Num, std::intmax_t Den>
+struct tag {
+	using id        = ID;
+	using is_no_tag = std::is_same<id,void>;
+	using exponent  = typename std::conditional<is_no_tag::value, exponent<0,1>, exponent<Num,Den>>::type;
+};
+
+/**
+ * @brief No tag
+ *
+ * This signifies that a quantity has no tag
+ */
+using no_tag = tag<void,0,1>;
+
+namespace detail {
+
+template<typename ID, std::intmax_t N=0, std::intmax_t D=0>
+struct create_tag                                               {using type = tag<ID,N,D>;};
+template<             std::intmax_t N  , std::intmax_t D  >
+struct create_tag<void,N,D>                                     {using type = no_tag;};
+template<typename ID                   , std::intmax_t D  >
+struct create_tag<ID,0,D>                                       {using type = no_tag;};
+template<                                std::intmax_t D  >
+struct create_tag<void,0,D>                                     {using type = no_tag;};
+template<typename ID, std::intmax_t N  , std::intmax_t D  >
+struct create_tag<tag<ID,N,D>,0,0>                              {using type = typename create_tag<ID,N ,D >::type;};
+template<typename ID, std::intmax_t N1 , std::intmax_t D1
+                    , std::intmax_t N2 , std::intmax_t D2 >
+struct create_tag<tag<ID,N1,D1>,N2,D2>                          {using type = typename create_tag<ID,N1,D1>::type;};
```

Comment:
this looks like a no-op to me, I would expect `create_tag<tag<ID,N1,D1>,N2,D2>::type` to be `tag<ID,N1,D1>`.

Am I missing something?